sample/composeApp/src/commonMain/kotlin/com/mifos/passcode/sample/chooseAuthOption/ChooseAuthOptionScreenViewmodel.kt

```diff
@@ -0,0 +1,54 @@
+package com.mifos.passcode.sample.chooseAuthOption
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import auth.deviceAuth.RegistrationResult
+import com.mifos.passcode.auth.deviceAuth.PlatformAuthenticationProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+
+class ChooseAuthOptionScreenViewmodel(
+    private val chooseAuthOptionRepository: ChooseAuthOptionRepository,
+) : ViewModel() {
+
+    private val _registrationResult = Channel<RegistrationResult?>()
+    val registrationResult = _registrationResult.receiveAsFlow()
+
+    fun setRegistrationResultNull() {
+        _registrationResult.trySend(null)
+    }
+
+    fun registerUser(
+        platformAuthenticationProvider: PlatformAuthenticationProvider,
+        userID: String = "",
+        userEmail: String = "",
+        displayName: String = ""
+    ) {
+        viewModelScope.launch(Dispatchers.Main) {
```

Comment:
when we use `viewmodelScope.launch` by default it runs in Main thread no need to reannounce it: